tests/lib/rules/forward-ref-uses-ref.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * @fileoverview Ensure symmetric naming of setState hook value and setter variables
+ * @author Duncan Beevers
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/forward-ref-uses-ref');
+const parsers = require('../../helpers/parsers');
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module',
+  },
+});
+
+const tests = {
+  valid: parsers.all([
+    {
+      code: `
+        import { forwardRef } from 'react'
+        forwardRef((props, ref) => {
+          return null
+        })
+      `,
+    },
+    {
+      code: `
+        import { forwardRef } from 'react'
+        forwardRef((props, ref) => null)
+      `,
+    },
+    {
+      code: `
+        import { forwardRef } from 'react'
+        forwardRef(function (props, ref) {
+          return null
+        })
+      `,
+    },
+    {
+      code: `
+        import { forwardRef } from 'react'
+        forwardRef(function Component(props, ref) {
+          return null
+        })
+      `,
+    },
+    {
+      code: `
+        import * as React from 'react'
+        React.forwardRef((props, ref) => {
+          return null
+        })
+      `,
+    },
+    {
+      code: `
+        import * as React from 'react'
+        React.forwardRef((props, ref) => null)
+      `,
+    },
+    {
+      code: `
+        import * as React from 'react'
+        React.forwardRef(function (props, ref) {
+          return null
+        })
+      `,
+    },
+    {
+      code: `
+        import * as React from 'react'
+        React.forwardRef(function Component(props, ref) {
+          return null
+        })
+      `,
+    },
+    {
+      code: `
+        import * as React from 'react'
+        function Component(props) {
+          return null
+        }
+      `,
+    },
+  ]),
+  invalid: parsers.all([
+    {
+      code: `
+        import { forwardRef } from 'react'
+        forwardRef((props) => {
+          return null
+        })
+      `,
+      errors: [
+        {
+          message:
+            'forwardRef is used with this component but no ref parameter is set',
+          suggestions: [
+            {
+              messageId: 'addRefParameter',
+              output: `
+        import { forwardRef } from 'react'
+        forwardRef((props, ref) => {
+          return null
+        })
+      `,
```

Comment:
```suggestion
             import { forwardRef } from 'react';
             forwardRef((props, ref) => {
               return null;
             });
           `,
```

Reply:
The indentation here is set up to match with the input, should I indent the input in lines 100-105 as well?

Reply:
ah, good call, this is fine as is